src/dvc_studio_client/post_live_metrics.py

```diff
@@ -55,17 +58,99 @@ def get_studio_repo_url() -> Optional[str]:
 
 def get_studio_token_and_repo_url(studio_token=None, studio_repo_url=None):
     studio_token = studio_token or getenv(DVC_STUDIO_TOKEN) or getenv(STUDIO_TOKEN)
-    if studio_token is None:
+    """Get studio token and repo_url. Kept for backwards compatibility."""
+    config = get_studio_config(
+        studio_token=studio_token, studio_repo_url=studio_repo_url
+    )
+    return config["token"], config["repo_url"]
+
+
+def get_studio_config(
```

Comment:
No need to block this P.R. but this logic now feels (probably before too) like would be better extracted to a separate file.

Reply:
It probably should mean moving everything other than `post_live_metrics` out of this file, right? I'd be hesitant to do that because I'm trying to keep it backwards compatible. What do you think?

Reply:
@daavoo Ping. Even though it's approved, I'd like to get your take before merging. Does it make sense as is, or do you still think something should be moved to a different module?